### DISCUSSION TITLE
Support for downloading different audio types?

### DISCUSSION THREAD

**Discussion by nednoodlehead** [DzenanJupic/rustube#56] [Ideas]
It would be nice if we had support for downloading audio via `best_audio()` could have some option to return the video as a `wav` or `mp3` version. Not sure if this is some limitation with the YouTube Api or not, but it would be really nice. And I believe `pytube` downloads as `mp3` by default, so it must be possible.

I have been trying to use the files with other rust audio libraries like `symphonia` and there is no support for the `opus` codec. Which makes using symphonia and rustube in the same project sort of complicated.

**Comment by DzenanJupic** [comment]
Hey @nednoodlehead, thanks for the feedback and sorry for the late answer.

If I'm not mistaken, `pytube` uses `ffmpeg` for doing that. At this point in time `rustube` does not do anything similar. It just provides raw access to the individual streams YouTube offers.

You can however try to filter out the types of audio formats you don't like:
```rust
let video: Video;
let audio_stream: &Stream = video
    .streams()
    .iter()
    .filter(|stream| stream.includes_audio_track && !stream.includes_video_track) // Get only the audio only streams
    .filter(|stream| stream.mime.subtype() == "mpeg") // Filter for MPEG streams (or whatever you're looking for)
    .max_by_key(|stream| stream.bitrate) // Get the best quality stream of all remaining
    .expect("No streams match the filters");
```

For converting audio streams of one type to another, you can always fall back to [`ffmpeg`](https://ffmpeg.org/download.html). I don't know a ton about it though, so I cannot help you there.
